src/message.rs

```diff
@@ -942,13 +942,33 @@ impl<'a> MessageBuilder<'a> {
             None => None,
         };
 
+        let mut timestamp = Utc::now();
+        let client_guard = crate::client::get_client(self.client_options).await?;
+        let client = client_guard.read().await;
+        let metadata = client.get_message().metadata(&self.id).await;
+        if metadata.is_ok() {
+            timestamp = match metadata.unwrap().referenced_by_milestone_index {
+                Some(ms_index) => {
+                    let ms = client.get_milestone(ms_index).await;
```

Comment:
@melatron can you add a cache mechanism in here? ideally we don't want to query the milestone API for each milestone index

assume you queried a milestone with index 464
and the next message was referenced by milestone 476
we can assume that the second message's timestamp is `first_message_timestamp + (476-464)*10s` since timestamps are issued every 10 seconds

of course that's not precise, so we can use that mechanism just on an specific interval, e.g. 100 indexes (so if another message has index 598, we query the node again and cache that info)

Reply:
This sounds like a good optimisation I will implement it.